### PR TITLE
feat: add Advanced settings for offchain spaces.

### PR DESCRIPTION
**Pull request by Sekhmet**
### Summary

Adding advanced settings page for offchain spaces with following options:
- Sub-spaces (parent and child spaces)
- Terms of services
- Custom domain
- Hide from homepage
- Delete space

Had to make some adjustments from the design, as not everything was clear:
- Not extraneous modals for parent-space, sub-space, custom-domain (in designs we have input, but also modal that opens for you to type in, which I'm not sure makes sense, here I just use inputs directly avoiding modals.
- Designs didn't cover how already added sub-spaces should look like, so I improvised.
- No tooltip for custom domain input as we don't have it either on v1 so I didn't have text to use.

Closes: #

### How to test

1. Create new space (if you want to delete it, **you will lose ability to recreate space under that domain**).
2. Go to your space, enter settings, you can modify advanced settings.
3. You can delete your space.

### Screenshots

localhost_8080_ (9)

### CODE REVIEW

File: apps/ui/src/views/Space/Settings.vue [hunk 1]
```diff
@@ -489,9 +516,28 @@ watchEffect(() => setTitle(`Edit settings - ${props.space.name}`));
         />
       </teleport>
     </UiContainerSettings>
+    <UiContainerSettings
+      v-else-if="activeTab === 'advanced'"
+      title="Advanced"
+      description="Office ipsum you must be muted. Boy ocean define crank new."
```
Comment (Sekhmet) [inline apps/ui/src/views/Space/Settings.vue:522]: @bonustrack will need text for this section.

File: apps/ui/src/components/FormSpaceAdvanced.vue [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+      :error="formErrors.termsOfServices"
+    />
+  </div>
+  <h4 class="eyebrow mb-2 font-medium mt-4">Custom domain</h4>
+  <UiMessage
+    type="info"
+    :learn-more-link="'https://docs.snapshot.org/spaces/add-custom-domain'"
+  >
+    To setup a custom domain you additionally need to open a pull request on
+    github after you have created the space.
```
Comment (bonustrack) [inline apps/ui/src/components/FormSpaceAdvanced.vue:214]:
```suggestion
    GitHub after you have created the space.
```

**Review by bonustrack — COMMENTED** [review]
It should be possible to add both a main space, and a sub spaces, it's not one or the other.

Also ideally we shouldn't allow duplicate for sub spaces.

**Review by wa0x6e — CHANGES_REQUESTED** [review]
Screenshot 2024-09-21 at 18 03 30

Disabled danger button should have disabled border color

File: apps/ui/src/components/Ui/Message.vue [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <div class="leading-5">
+      <slot />
+      <a
+        v-if="learnMoreLink"
+        :href="learnMoreLink"
+        target="_blank"
+        class="flex items-center text-skin-link mt-1 gap-1 w-fit"
+      >
+        Learn more
+        <IH-arrow-top-right-on-square />
```
Comment (wa0x6e) [inline apps/ui/src/components/Ui/Message.vue:25]: This is indeed from Figma, but on v2, we're now using the 45deg arrow without the square box for external link.

Better to follow the v2 format

**Review by wa0x6e — CHANGES_REQUESTED** [review]
The parent and child space input form should have a loading indicator

Screenshot 2024-09-21 at 18 24 54

**Review by wa0x6e — CHANGES_REQUESTED** [review]
Dirtying any fields in the new ADVANCED tab is not showing the bottom toolbar with SAVE button

**Review by wa0x6e — COMMENTED** [review]
Should prevent duplicate in child spaces

Screenshot 2024-09-21 at 18 30 22

File: apps/ui/src/composables/useActions.ts [hunk 4]
```diff
@@ -541,6 +541,23 @@ export function useActions() {
...
 
+  async function deleteSpace(space: Space) {
+    if (!web3.value.account) {
+      await forceLogin();
+      return null;
+    }
+
+    const network = getNetwork(space.network);
+    if (!network.managerConnectors.includes(web3.value.type as Connector)) {
+      throw new Error(`${web3.value.type} is not supported for this actions`);
```
Comment (wa0x6e) [inline apps/ui/src/composables/useActions.ts:552]:
```suggestion
      throw new Error(`${web3.value.type} is not supported for this action`);
```

**Review by wa0x6e — CHANGES_REQUESTED** [review]

**Comment by wa0x6e** [comment]
> It should be possible to add both a main space, and a sub spaces, it's not one or the other.
> 
> Also ideally we shouldn't allow duplicate for sub spaces.

Seems like on v1, it's one or the other, both are not possible

**Comment by wa0x6e** [comment]
Screenshot 2024-09-21 at 19 05 44

Space between checkbox and label seen insufficient

**Comment by Sekhmet** [comment]
> The parent and child space input form should have a loading indicator

Updated it to have similar space validation UX as v1.

> Dirtying any fields in the new ADVANCED tab is not showing the bottom toolbar with SAVE button

Should be fixed now.

> Space between checkbox and label seen insufficient

Followed designs there, but I agree this doesn't look good, increased it to `gap-2`.

**Comment by Sekhmet** [comment]
> It should be possible to add both a main space, and a sub spaces, it's not one or the other.

As Wan mentioned this replicates v1 UI that prevents configuring both parent and sub-spaces at the same time. I can remove that validation if we confirm that's what we want.

**Review by wa0x6e — CHANGES_REQUESTED** [review]
The domain name field is missing the length validation

Screenshot 2024-09-23 at 18 39 50

Putting something over the char limit does not show any errors.

And it's not on v1, but maybe it's time to also put a proper domain name validation on this field, since we'll be using it automatically for the whitelabel

On v1, there's a human manual validation, but we're moving away from that for v2

**Review by wa0x6e — CHANGES_REQUESTED** [review]
When putting an invalid space name in the parent space, it shows an error.

But it we change tab, and go back to advanced, the error state is gone.

There seems to be a query to check if the space is valid (loading icon on the input on tab change), but the error state is not being restored

**Review by wa0x6e — COMMENTED** [review]
Screenshot 2024-09-23 at 19 01 42

Shouldn't the message be different ?

**Comment by Sekhmet** [comment]
> When putting an invalid space name in the parent space, it shows an error.
> But it we change tab, and go back to advanced, the error state is gone.
> There seems to be a query to check if the space is valid (loading icon on the input on tab change), but the error state is not being restored

It's broken app-wide, we should fix it properly, but not really belongs in this PR, we can workaround it by using `v-show` which is not ideal (used it there for now).
https://github.com/snapshot-labs/sx-monorepo/issues/573

**Comment by Sekhmet** [comment]
> And it's not on v1, but maybe it's time to also put a proper domain name validation on this field, since we'll be using it automatically for the whitelabel

What you consider proper domain name validation? I think anything "proper" would be out of scope of this PR.

**Review by wa0x6e — COMMENTED** [review]
When reaching the max number of sub-spaces, the sub-space form is still accepting value, and validating space name.

But does not show any error when the space name is valid. I think it should show the error about reaching the max number of sub-spaces

**Comment by wa0x6e** [comment]
> out of scope of this PR.

For now, just `alphanum`, `.`, `_` and `-` will cover 100% of our existing list (https://github.com/snapshot-labs/snapshot-spaces/blob/master/spaces/domains.json) we can optimize later

**Review by wa0x6e — COMMENTED** [review]
With some unsaved valid changes, navigating between non-advanced tab is keeping the bottom toolbar (with the save button) always shown.

But when navigating to advanced tab, the toolbar disappear for a spit-second, before reappearing

**Review by bonustrack — APPROVED** [review]
tACK

**Review by wa0x6e — CHANGES_REQUESTED** [review]
Looks like the dirty state of the sub-spaces is only looking at the number of spaces.

Example:

- My space already 3 sub-spaces
- I remove 1 space -> SAVE button shows up
- I add another different space -> SAVE button disappear

**Review by wa0x6e — APPROVED** [review]
tAck

**Pull request merged**